src/test/java/by/home/controller/DirectionTest.java

```diff
@@ -0,0 +1,33 @@
+package by.home.controller;
+
+import by.home.entity.Circle;
+import by.home.entity.Point;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class DirectionTest {
+
+    private final static String TEST_FILE_PATH = "./src/test/resources/testFile.txt";
+    private final static List<Circle> EXPECTED_CIRCLES = Arrays.asList(
+            new Circle(new Point(2, 2), 5),
+            new Circle(new Point(5, 10), 10),
+            new Circle(new Point(0, 0), 2),
+            new Circle(new Point(-8, 8), 5));
+
+    @Test
+    public void testReadShouldReturnCirclesList() throws DataException, RadiusException {
+
+        //given
```

Comment:
DataReader  reader = Mockito.mock(DataReader.class);
when(reader.read(anyString())).thenReturn(Arrays.asList("1,1,0","0,0,1")); //here you fake the read behavior
//do each for each class that you pass in constructor
... 
Direction direction = new Direction(reader, ....);